### PR TITLE
employer_profile_builder spec fix

### PR DESCRIPTION
**Pull request by jacobkagon**
N/A

### CODE REVIEW

**Review by markgoho — COMMENTED** [review]
Why did this code just start failing after 5 years?

**Review by harshaellanki — CHANGES_REQUESTED** [review]
this is not a correct fix, this will fail again in december

File: spec/models/sponsored_benefits/employer_profile_builder_spec.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 module SponsoredBenefits
   RSpec.describe BenefitApplications::EmployerProfileBuilder, type: :model do
-    let(:effective_period_start_on) { TimeKeeper.date_of_record.end_of_month + 1.day + 1.month }
+    let(:effective_period_start_on) { TimeKeeper.date_of_record.end_of_month + 1.day }
```
Comment (saikumar9) [inline spec/models/sponsored_benefits/employer_profile_builder_spec.rb:6]: @jacobkagon Why can't we make effective_period_start_on as TimeKeeper.date_of_record.beginning_of_month? and in your 'it' block, you can expect different things if the system is under OE or non-OE.

**Comment by harshaellanki** [comment]
> Why did this code just start failing after 5 years?

@markgoho it failed on jan 1st and we refactored it to use shared context dates. fix went in this [PR](https://github.com/ideacrew/enroll/pull/1535)

**Review by harshaellanki — APPROVED** [review]

**Pull request merged**